Border for `st.columns`

mayagbarnes opened the pull request:
## Describe your changes
PR adds `border` param for `st.columns` - default is `False` (current behavior)

* Closes #8547 

## Testing Plan
- Unit Tests: ✅ 
- E2E Tests: ✅ 
- Manual Testing: ✅ 

-----

**Light Comparison**
<img width="500" alt="Screenshot 2024-11-25 at 2 27 04 PM" src="https://github.com/user-attachments/assets/86d86523-93bb-4439-be5e-617eef30dad3">

**Dark Comparison**
<img width="500" alt="Screenshot 2024-11-25 at 2 27 19 PM" src="https://github.com/user-attachments/assets/6693766a-2b65-4431-bb1b-9ccd3d706951">

mayagbarnes committed:
Add border param & tests

mayagbarnes committed:
Test

mayagbarnes committed:
Update snapshots

raethlein commented on e2e_playwright/__snapshots__/linux/st_columns_test/st_columns-column_gap_large[chromium].png:
Just for sanity checking: There is no actual visual change here, right? Probably just something because of the DOM structure or so

mayagbarnes replied:
Correct! Did some digging to check and didnt find any real difference

raethlein commented on proto/streamlit/proto/Block.proto:
```diff
@@ -56,6 +56,7 @@ message Block {
       BOTTOM = 2;
     }
     VerticalAlignment vertical_alignment = 3;
+    bool border = 4;
```
Similar to the comment [here](https://github.com/streamlit/streamlit/pull/9928#discussion_r1867673984): I would indicate in the variable name that this is of boolean type

sfc-gh-dmatthews commented on lib/streamlit/elements/layouts.py:
```diff
@@ -216,6 +217,10 @@ def columns(
             default is ``"top"``.
 
+        border : bool
+            Whether to show a border. If ``True``, borders are shown around each column.
+            The default is ``False``.
```
```suggestion
            Whether to show a border around the metric container. If this is
            ``False`` (default), no border is shown. If this is ``True``, a
            border is shown on each column.
```

mayagbarnes committed:
Address PR comments

mayagbarnes committed:
Merge branch 'develop' into column-border

sfc-gh-dmatthews committed:
Column borders docstring (#9962)

## Describe your changes
Add border example to st.columns

---
**Contribution License Agreement**
By submitting this pull request you agree that all contributions to this
project are made under the Apache 2.0 license.

mayagbarnes committed:
Merge branch 'develop' into column-border

mayagbarnes committed:
Snapshot flakiness
